torch/_inductor/lowering.py

```diff
@@ -2362,9 +2375,10 @@ def is_aligned(x):
         return ir.ExternKernel.require_stride_order(arg, stride_order)
 
     args = tuple(
-        apply_constraint(arg, fx_arg) for arg, fx_arg in zip(args, fx_node.args)
+        apply_constraint(idx, arg, fx_arg)
+        for idx, (arg, fx_arg) in enumerate(zip(args, fx_node.args))
     )
-    kwargs = {k: apply_constraint(v, fx_node.kwargs[k]) for k, v in kwargs.items()}
+    kwargs = {k: apply_constraint(-1, v, fx_node.kwargs[k]) for k, v in kwargs.items()}
```

Comment:
Should this be something like

```Python
kwargs = {
    k: apply_constraint(len(args) + idx, v, fx_node.kwargs[k]) 
    for idx, (k, v) in enumerate(kwargs.items())
}
```

Im not sure how the args are passed to the backwards func there might not be kwargs but probably good just incase

Reply:
I'm worries that if the order of keys in kwargs does not match the order in the op schema, we may mess things up. Right now I only see the out/grad_out appearing as positional arguments. I'm not sure how fragile this is though